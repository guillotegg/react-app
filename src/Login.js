import React, { Component } from 'react';
import { Input } from '@progress/kendo-react-inputs';
import { Button } from '@progress/kendo-react-buttons';
import { Window } from '@progress/kendo-react-dialogs';

class Login extends Component {

    constructor(props) {
        super(props);
        
        this.state = { 
            username:'',
            password:''
        };

        this.updateUserName = this.updateUserName.bind(this);
        this.updatePassword = this.updatePassword.bind(this);
        this.login = this.login.bind(this);
    }

    updateUserName = function name(evt) {
        this.setState({
            username: evt.target.value
        });
    }

    updatePassword = function name(evt) {
        this.setState({
            password: evt.target.value
        });
    }
  
    login(){
       if (this.state.username==="demo" && this.state.password==="demo"){
            return this.props.history.push('/App');
       }
    }

    render() {
        return <div>
            <Window title="Log In" resizable={false} draggable={false} top={0} left={0} initialWidth={500} initialHeight={300}>
                <div class="login">
                    <div>
                        <Input label="username" onChange={this.updateUserName} />
                    </div>
                    <div>
                        <Input type="password" label="password" onChange={this.updatePassword} />
                    </div>
                    <br />
                    <div>
                        <Button onClick={this.login}>Log In!</Button>                    
                    </div>
                </div>
            </Window>
        </div>
    }
}

export default Login;