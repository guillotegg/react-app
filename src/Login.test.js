import React from 'react';
import { configure, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Login from './Login';

describe('Login', () => {
    
    test('should correctly render Login', () => {
        const wrapper = shallow(<Login />);
        expect(wrapper).toMatchSnapshot();
    });
    
    let result;
    test('should render a div when bound to null', () => {
        result = shallow(<Login data={null} />);
        expect(result.type()).toEqual('div');
    })
})
