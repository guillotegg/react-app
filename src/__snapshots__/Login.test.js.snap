// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should correctly render Login 1`] = `ShallowWrapper {}`;
